exercises/Stephen_Smith/ex02_oo_basics/include/ShapeSorter.h

```diff
@@ -0,0 +1,34 @@
+#ifndef SHAPESORTER_H_
+#define SHAPESORTER_H_
+
+#include "Shapes.h"
```

Comment:
One thing to be careful of in general is only including what you use so that we can make sure compile times don't build up. In this file only `vector`, `string` would be required. For `BaseShape` we actually do something called a [forward declaration](https://www.geeksforgeeks.org/what-are-forward-declarations-in-c/) so we don't need to include it in the header at all - it would just need to be included in the .cpp file.

Reply:
Okay useful to know, I'll keep this in mind, and I have adjusted the file ShapeSorter.h accordingly. Am I right in saying you cannot forward declare the base class within the other shapes, e.g Triangle?

Reply:
That's correct. You can only forward declare something when you don't need to know the size of the actual class. In `ShapeSorter` it only uses a pointer and they are always the same size. When inheriting the compiler needs to know the layout of the base class so the include will be needed.